geode-core/src/upgradeTest/java/org/apache/geode/security/AuthExpirationDUnitTest.java

```diff
@@ -70,15 +84,70 @@
       .withSecurityManager(ExpirableSecurityManager.class)
       .withRegion(RegionShortcut.REPLICATE, "region");
 
+
+  private ClientVM clientVM;
+
   @Test
-  public void clientShouldReAuthenticateWhenCredentialExpiredAndOperationSucceed()
-      throws Exception {
+  public void clientWithNoUserRefreshWillNotSucceed() throws Exception {
     int serverPort = server.getPort();
     ClientVM clientVM = lsRule.startClientVM(0, clientVersion,
         c -> c.withProperty(SECURITY_CLIENT_AUTH_INIT, UpdatableUserAuthInitialize.class.getName())
             .withPoolSubscription(true)
             .withServerConnection(serverPort));
 
+    clientVM.invoke(() -> {
+      UpdatableUserAuthInitialize.setUser("user1");
+      ClientCache clientCache = ClusterStartupRule.getClientCache();
+      ClientRegionFactory<Object, Object> clientRegionFactory =
+          clientCache.createClientRegionFactory(ClientRegionShortcut.PROXY);
+      Region<Object, Object> region = clientRegionFactory.create("region");
+      region.put(0, "value0");
+    });
+
+    // expire the current user
+    ExpirableSecurityManager securityManager = getSecurityManager();
+    securityManager.addExpiredUser("user1");
+
+    // if client, even after getting AuthExpiredExpiration, still sends in
+    // old credentials, the operation will fail (we only try re-authenticate once)
+    // this test makes sure no lingering old credentials will allow the operations to succeed.
+    clientVM.invoke(() -> {
+      ClientCache clientCache = ClusterStartupRule.getClientCache();
+      Region<Object, Object> region = clientCache.getRegion("region");
+      doPutAndExpectFailure(region, 100);
+    });
+
+    Region<Object, Object> region = server.getCache().getRegion("/region");
+    assertThat(region.size()).isEqualTo(1);
+    Map<String, List<String>> authorizedOps = securityManager.getAuthorizedOps();
+    Map<String, List<String>> unAuthorizedOps = securityManager.getUnAuthorizedOps();
+    assertThat(authorizedOps.keySet().size()).isEqualTo(1);
+    assertThat(authorizedOps.get("user1")).asList().containsExactly("DATA:WRITE:region:0");
+    assertThat(unAuthorizedOps.keySet().size()).isEqualTo(1);
+  }
+
+  private static void doPutAndExpectFailure(Region<Object, Object> region, int times) {
+    for (int i = 1; i < times; i++) {
+      try {
+        region.put(1, "value1");
+        fail("Exception expected");
+      } catch (Exception e) {
+        assertThat(e).isInstanceOf(ServerOperationException.class);
+        assertThat(e.getCause()).isInstanceOfAny(AuthenticationFailedException.class,
+            AuthenticationRequiredException.class);
+      }
```

Comment:
This can be simplified to:
```
      assertThatThrownBy(() -> region.put(1, "value1"))
          .isInstanceOf(ServerOperationException.class)
          .getCause().isInstanceOfAny(AuthenticationFailedException.class,
          AuthenticationRequiredException.class);
```